exercises/edward_brown/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,45 @@
+#include "FileHelpers.h"
+#include "MakeWordFrequencyMap.h"
+#include "WordFrequencyTable.h"
+#include <iostream>
+
+/** Ensures the existance of at least one command line argument to be used as the input
+ *  file name.
+ *  @param ArgumentCount The value of the argc parameter passed to main.
+ *  @return True if the minimum number of arguments or greater are present.
+ */
+bool hasRequiredCommandLineArguments(int ArgumentCount) {
+  auto constexpr MINIMUM_COMMAND_LINE_ARGUMENTS = 1;
+  return MINIMUM_COMMAND_LINE_ARGUMENTS <= ArgumentCount;
+}
+
+/// Indicates to the user that the output has been created and written successfully.
+void indicateSuccess() {
+  std::cout << "Successfully wrote word frequencies to results.txt."
+            << std::endl;
+}
+
+/// Indicates to the user that no input files were provided.
+void noInputFiles() { std::cerr << "No input file specified." << std::endl; }
+
+int main(int argc, char **argv) {
+  if (hasRequiredCommandLineArguments(argc)) {
+    try {
+      auto const InputFileName = std::string(argv[1]);
+      auto const InputString = readInputFile(InputFileName);
```

Comment:
I changed the approach in f4a8b5b8e33a8c2193f78109ff5b7785e29d4b5d which doesn't read the entire file at once but also not line by line. The alternative would be to make more significant changes to the interfaces of `forEachWord` and `makeWordFrequencyMap` or even write a custom file iterator.